src/serde.rs

```diff
@@ -0,0 +1,111 @@
+#[cfg(feature = "serialize")]
+use serde::{Serialize, Serializer};
+#[cfg(feature = "deserialize")]
+use serde::{Deserialize, Deserializer};
+use crate::{RefCnt, ArcSwapAny, Strategy};
+
+#[cfg(feature = "serialize")]
+/// # Examples
+///
+/// ```rust
+/// use arc_swap::ArcSwap;
+///
+/// #[derive(serde::Serialize)]
+/// struct Foo {
+///     field0: usize,
+///     field1: String,
+/// }
+///
+/// let data = Foo {
+///     field0: 123,
+///     field1: "123".to_owned(),
+/// };
+///
+/// let data_json = serde_json::to_string(&data).unwrap();
+/// println!("{}", data_json);
+/// ```
+impl<T, S> Serialize for ArcSwapAny<T, S>
+    where
+        T: RefCnt + Serialize,
+        S: Strategy<T>,
+{
+    fn serialize<Ser: Serializer>(&self, serializer: Ser) -> Result<Ser::Ok, Ser::Error> {
+        self.load().serialize(serializer)
+    }
+}
+
+#[cfg(feature = "deserialize")]
+/// # Examples
+///
+/// ```rust
+/// use arc_swap::ArcSwap;
+///
+/// #[derive(serde::Deserialize)]
+/// struct Foo {
+///     field0: usize,
+///     field1: String,
+/// }
+///
+/// let data = serde_json::from_str::<ArcSwap<Foo>>(r#"{"field0":123,"field1":"123"}"#).unwrap();
+/// println!("{:?}", data);
+/// ```
+impl<'de, T, S> Deserialize<'de> for ArcSwapAny<T, S>
+    where
+        T: RefCnt + Deserialize<'de>,
+        S: Strategy<T> + Default,
+{
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        Ok(Self::from(T::deserialize(deserializer)?))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{ArcSwapOption, ArcSwap};
```

Comment:
I very much prefer the `use super::*` approach in tests. That will also allow you to just use `#[derive(Serialize)]` below instead of `#[derive(serde::Serialize)]`.

Reply:
It has no sense, because I don't use `ArcSwapOption` and `ArcSwap` in `super` and `#[derive(Serialize)]` / `#[derive(Deserialize)]` from `serde_derive` package.